subaccounts.go

```diff
@@ -128,33 +122,35 @@ func (c *Client) SubaccountUpdate(s *Subaccount) (res *Response, err error) {
 
 // SubaccountUpdateContext is the same as SubaccountUpdate, and it allows the caller to provide a context
 func (c *Client) SubaccountUpdateContext(ctx context.Context, s *Subaccount) (res *Response, err error) {
-	if s.ID == 0 {
-		err = fmt.Errorf("Subaccount Update called with zero id")
+	if s == nil {
+		err = errors.New("Subaccount Update called with nil Subaccount")
+	} else if s.ID == 0 {
+		err = errors.New("Subaccount Update called with zero id")
 	} else if len(s.Name) > 1024 {
-		err = fmt.Errorf("Subaccount name may not be longer than 1024 bytes")
+		err = errors.New("Subaccount name may not be longer than 1024 bytes")
+	} else if len(s.IPPool) > 20 {
+		err = errors.New("Subaccount ip pool may not be longer than 20 bytes")
 	} else if s.Status != "" {
 		found := false
-		for _, v := range validStatuses {
+		for _, v := range SubaccountStatuses {
 			if s.Status == v {
 				found = true
 			}
 		}
 		if !found {
-			err = fmt.Errorf("Not a valid subaccount status")
+			err = errors.New("Not a valid subaccount status")
 		}
 	}
 
 	if err != nil {
 		return
```

Comment:
Wouldn't it be better to return the error so the user has a change to know what the problem was?